src/Core/Silk.NET.Core/Native/SilkMarshal.cs

```diff
@@ -142,9 +142,8 @@ public static int GetMaxSizeOf(string? input, NativeStringEncoding encoding = Na
             => encoding switch
             {
                 NativeStringEncoding.BStr => -1,
-                NativeStringEncoding.LPStr => ((input?.Length ?? 0) + 1) * Marshal.SystemMaxDBCSCharSize + 1,
-                NativeStringEncoding.LPTStr => (input is null ? 0 : Encoding.UTF8.GetMaxByteCount(input.Length)) + 1,
-                NativeStringEncoding.LPUTF8Str => (input is null ? 0 : Encoding.UTF8.GetMaxByteCount(input.Length)) + 1,
+                NativeStringEncoding.LPStr or NativeStringEncoding.LPTStr or NativeStringEncoding.LPUTF8Str
+                    => (input is null ? 0 : Encoding.UTF8.GetMaxByteCount(input.Length)) + 1,
```

Comment:
isnt LPTStr different if `UNICODE` is defined? what does silk prefer for windows functions?
(eg. LPTStr is a pointer to a 16bit number when `UNICODE` is defined but a pointer to an 8bit number if not)

Reply:
You're right however we've the LPStr/LPTStr/LPUTF8Str enums are just for compatibility and aren't really intended to be used (we're using them here out of habit :/). The Ansi/Auto/Uni/UTF8 ones are designed to be used.

Silk.NET has never observed this LPTStr behaviour and may be confusing to users if we changed this. Admittedly this was an oversight, but a 3 year old one that noone seems to have kicked up a fuss about